### PR TITLE
Git optimization feature documentation

### PR DESCRIPTION
**Pull request by maksymvyhovskyi**

### CODE REVIEW

File: content/en/docs/refguide/modeling/menus/version-control-menu/_index.md [hunk 1]
```diff
@@ -62,11 +62,28 @@ The **Reverse Merge Changes** option allows locally rolling back changes that we
...
+## 12 Optimize Storage Repository {#optimize-storage}
+
+{{% alert color="info" %}}
+
+The **Optimize Repository Storage…** option is only available in Git. 
+
+{{% /alert %}}
+
+The **Optimize Repository Storage…** option allows you to perform a number of housekeeping tasks in the current Git repository. Git repositories can accumulate various types of garbage over the time, which may lead to possible performance bottlenecks when using version control features.
+When you optimize the repository storage, the file revisions are compressed to reduce disk space and increase performance and unreachable Git storage objects are removed. For more information on storage optimization, see [Git Storage Optimization](/refguide/git-storage-optimization/).
```
Comment (maksymvyhovskyi) [inline content/en/docs/refguide/modeling/menus/version-control-menu/_index.md]: The link should be /refguide/gid-storage-optimization-dialog/ I guess.

File: content/en/docs/refguide/modeling/menus/version-control-menu/_index.md [hunk 2]
```diff
@@ -62,11 +62,28 @@ The **Reverse Merge Changes** option allows locally rolling back changes that we
...
+
+{{% alert color="info" %}}
+
+The operation might some time. While optimization is in progress, most of version control commands, such as committing and updating, are disabled.
```
Comment (maksymvyhovskyi) [inline content/en/docs/refguide/modeling/menus/version-control-menu/_index.md]: might TAKE ?

File: content/en/docs/refguide/version-control/git-storage-optimization-dialog.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+Finishing the optimization process does not close the **Optimize Repository Storage** pop-up window automatically. You have to do this manually. You can also close the pop-up window during the optimization process, this does not affect the process.
+{{% /alert %}}
+
+## 3 Automatic Repository Optimization {#automatic-optimization}
+
+To optimize Git repository storage automatically and regularly, do the following:
+
+1. Open the **Edit** menu > **Preferences** > **Version Control** tab. 
+2. In the **Git** section, toggle the **Enable automatic repository optimization** option.
+3. In the **Number of commits**, specify the maximum number of commits that should be reached to start an optimization process in background (for more information, see [Preferences](/refguide/preferences-dialog/).)
```
Comment (maksymvyhovskyi) [inline content/en/docs/refguide/version-control/git-storage-optimization-dialog.md]: ... **minimum** number of commits ...

File: content/en/docs/refguide/version-control/git-storage-optimization-dialog.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+4. Click **OK**.
+4. Studio Pro keeps track of user's activities now, however, reaching or overcoming the limit does not mean the process starts immediately. To trigger the process, either push (**Version Control** > **Push**) or commit (**Version Control** > **Commit** with **Also push changes to the remote server** option on).
+
+As soon as optimization is completed, Studio Pro resets the counter of commit messages to zero.
```
Comment (maksymvyhovskyi) [inline content/en/docs/refguide/version-control/git-storage-optimization-dialog.md]: ... of **commits** to zero

File: content/en/docs/refguide/version-control/git-storage-optimization-dialog.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+{{% alert color="info" %}}
+Disabling the feature does not stop the repository commit counter (you can find it in the application folder: open `.git/config` file and find `commits-since-gc` parameter section), it will continue counting the commits. Note this when you enable the automatic repository optimization back again.
```
Comment (maksymvyhovskyi) [inline content/en/docs/refguide/version-control/git-storage-optimization-dialog.md]: and find `commits-since-gc` parameter)    i.e. remove **section**

**Review by MariaShaposhnikova — APPROVED** [review]

**Pull request merged**